src/transport/raw/tests/TestTCP.cpp

```diff
@@ -111,81 +181,171 @@ void CheckSimpleInitTest6(nlTestSuite * inSuite, void * inContext)
 void CheckMessageTest(nlTestSuite * inSuite, void * inContext, const IPAddress & addr)
 {
     TestContext & ctx = *reinterpret_cast<TestContext *>(inContext);
+    TCPImpl tcp;
 
-    uint16_t payload_len = sizeof(PAYLOAD);
-
-    chip::System::PacketBufferHandle buffer = chip::System::PacketBufferHandle::NewWithData(PAYLOAD, payload_len);
-    NL_TEST_ASSERT(inSuite, !buffer.IsNull());
-
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    MockTransportMgrDelegate gMockTransportMgrDelegate(
+        inSuite, ctx, [](const uint8_t * message, size_t length, void * data) { return memcmp(message, data, length); },
+        const_cast<void *>(static_cast<const void *>(PAYLOAD)));
 
-    TCPImpl tcp;
+    gMockTransportMgrDelegate.InitializeMessageTest(tcp, addr);
+    gMockTransportMgrDelegate.SingleMessageTest(tcp, addr);
+    gMockTransportMgrDelegate.FinalizeMessageTest(tcp, addr);
+}
 
-    err = tcp.Init(Transport::TcpListenParameters(&ctx.GetInetLayer()).SetAddressType(addr.Type()));
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+void CheckMessageTest4(nlTestSuite * inSuite, void * inContext)
+{
+    IPAddress addr;
+    IPAddress::FromString("127.0.0.1", addr);
+    CheckMessageTest(inSuite, inContext, addr);
+}
 
-    MockTransportMgrDelegate gMockTransportMgrDelegate(inSuite);
-    TransportMgrBase gTransportMgrBase;
-    gTransportMgrBase.SetSecureSessionMgr(&gMockTransportMgrDelegate);
-    gTransportMgrBase.SetRendezvousSession(&gMockTransportMgrDelegate);
-    gTransportMgrBase.Init(&tcp);
+void CheckMessageTest6(nlTestSuite * inSuite, void * inContext)
+{
+    IPAddress addr;
+    IPAddress::FromString("::1", addr);
+    CheckMessageTest(inSuite, inContext, addr);
+}
 
-    ReceiveHandlerCallCount = 0;
+} // namespace
 
+// Generates a packet buffer or a chain of packet buffers for a single message.
+// `sizes[]` is a zero-terminates sequence of packet buffer sizes.
+// The first (or only) buffer will *additionally* include a total length field and PacketHeader;
+// except for these, the buffer contents is not initialized.
```

Comment:
Probably better to initialize to 0, in case we ever run these tests under valgrind or the like.